pkg/services/sqlstore/dashboard_provisioning.go

```diff
@@ -88,3 +89,28 @@ func UnprovisionDashboard(cmd *models.UnprovisionDashboardCommand) error {
 	}
 	return nil
 }
+
+func DeleteOrphanedProvisionedDashboard(cmd *models.DeleteOrphanedProvisionedDashboardsCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		var result []*models.DashboardProvisioning
+
+		convertedReaderNames := make([]interface{}, len(cmd.ReaderNames))
+		for index, readerName := range cmd.ReaderNames {
+			convertedReaderNames[index] = readerName
+		}
+
+		err := sess.NotIn("name", convertedReaderNames...).Find(&result)
+		if err != nil {
+			return err
+		}
+
+		for _, deleteDashCommand := range result {
+			err := deleteDashboard(&models.DeleteDashboardCommand{Id: deleteDashCommand.DashboardId}, sess)
+			if err != nil && err != models.ErrDashboardNotFound {
```

Comment:
I think it's better to detect an error type with [errors.Is](https://golang.org/pkg/errors/#Is), since it will handle wrapped errors:
```suggestion
			if err != nil && !errors.Is(err, models.ErrDashboardNotFound) {
```

Reply:
You right, it looks better.